src/index.js

```diff
@@ -1,30 +1,61 @@
 function onReady () {
-  const addTodoForm = document.getElementById( 'addTodoForm' );
-  const todoList = document.getElementById( 'todoList' );
-  const newTodoText = document.getElementById( 'newTodoText' );
+  initUi();
 
-  // <ul></ul>
-  todoList.textContent = '';
+  // it's a little confusing to see you do it without a state object
+  // because in the first week's activities, the todo array is nested inside
+  // state{}. Not a big deal, but might throw some people off. 
+  let todos = [];
 
-  addTodoForm.addEventListener( 'submit', event => {
-    event.preventDefault();
-    const title = newTodoText.value;
+  function createNewTodo ( title ) {
+    const newTodo = {title, complete: false}
 
-    // add it to the list
-    // create an li
-    // <li>{title}</li>
-    const newLi = document.createElement( 'li' );
-    newLi.textContent = title; // vs newLi.innerHTML
+  	todos.push(newTodo)
 
-    newLi.addEventListener( 'click', () => {
-      newLi.classList.toggle( 'todo--complete' );
-    });
+  	// The "state" changed, so re-draw the UI
+  	renderTheUi( newTodo )
+  	// I see where you're going here, esp. towards the React lifecycles. 
+  	// It's useful to see it built from scratch. 
+  	// Is this something that developed over time and React just formalized,
+  	// or is this a pattern React is championing and is just becoming defacto?
```

Comment:
More the latter; the idea is to make the UI a function of state (`state => ui`). A lot of frameworks build in this capability, but really React is the one that takes it all the way. Vue does a little too.